mapie/conformity_scores.py

```diff
@@ -0,0 +1,259 @@
+                raise ValueError(
+                    "The two functions get_conformity_scores and "
+                    "get_observed_value of the ConformityScore class "
+                    "are not consistent. "
+                    "The following equation must be verified: "
+                    "self.get_observed_value(y_pred, self.get_conformity_scores(y, y_pred)) == y. "  # noqa: E501
+                    f"The maximum conformity score is {max_conf_score}."
+                    "The eps attribute may need to be increased if you are "
+                    "sure that the two methods are consistent."
+                )
+
+
+class AbsoluteConformityScore(ConformityScore):
+    """Absolute conformity score.
+
+    The signed conformity score = y - y_pred.
+    The conformity score is symmetrical.
+
+    This is appropriate when the confidence interval is symmetrical and
+    its range is approximatively the same over the range of predicted values.
+    """
+
+    def __init__(self) -> None:
+        ConformityScore.__init__(self, True, consistency_check=False)
+
+    def get_signed_conformity_scores(
+        self,
+        y: ArrayLike,
+        y_pred: ArrayLike,
+    ) -> ArrayLike:
+        """
+        Compute the signed conformity scores from the predicted values
+        and the observed ones, from the following formula:
+        signed conformity score = y - y_pred
+        """
+        return np.subtract(y, y_pred)
+
+    def get_observed_value(
+        self, y_pred: ArrayLike, conformity_scores: ArrayLike
+    ) -> ArrayLike:
+        """
+        Compute the observed values from the predicted values and the
+        conformity scores, from the following formula:
+        signed conformity score = y - y_pred
+        <=> y = y_pred + signed conformity score
+        """
+        return np.add(y_pred, conformity_scores)
+
+
+class GammaConformityScore(ConformityScore):
+    """Gamma conformity score.
+
+    The signed conformity score = (y - y_pred) / y_pred.
+    The conformity score is not symmetrical.
+
+    This is appropriate when the confidence interval is not symmetrical and
+    its range depends on the predicted values.
+    """
+
+    def __init__(self) -> None:
+        ConformityScore.__init__(self, False, consistency_check=False)
+
+    def get_signed_conformity_scores(
+        self,
+        y: ArrayLike,
+        y_pred: ArrayLike,
+    ) -> ArrayLike:
+        """
+        Compute the signed conformity scores from the predicted values
+        and the observed ones, from the following formula:
+        signed conformity score = (y - y_pred) / y_pred
+        """
+        self._check_observed_data(y)
+        self._check_predicted_data(y_pred)
+        return np.divide(np.subtract(y, y_pred), y_pred)
+
+    def get_observed_value(
+        self, y_pred: ArrayLike, conformity_scores: ArrayLike
+    ) -> ArrayLike:
+        """
+        Compute the observed values from the predicted values and the
```

Comment:
As discussed in another comment, I'm not comfortable with the word "observed" since this function computes the distribution of predicted bounds.

Reply:
Done